pkg/controller/operators/catalog/operator.go

```diff
@@ -1269,26 +1246,44 @@ func (o *Operator) setSubsCond(subs []*v1alpha1.Subscription, condType v1alpha1.
 		cond.Reason = reason
 		cond.Message = message
 		if setTrue {
+			if cond.Status == corev1.ConditionTrue {
+				continue
+			}
```

Comment:
This change means that if the existing sub has:
```yaml
- type: Foo
  status: True
  reason: Hello
  message: World
```

And we call this function with `setSubsCond([]{sub}, "Foo", "True",  "Goodnight", "Moon")`, then the resulting condition will not have it's reason and message updated to "Goodnight" and "Moon", right?

That seems a little bit counter-intuitive to me.

Reply:
@joelanford that's a good catch. However, in this particular case, our objective for now is to optimize the number of times we hit the api server to update the subscriptions. And the only call site where `setSubsCond` is being called for do not have a use case for changing the `reason` and `message` of a condition. 

Which means getting rid of the check and letting the update go through regardless would be sacrificing current optimization for a potential future use case. We can always get rid of the check if a similar use case to the one you mentioned arises later on.

Reply:
What if the reason changes from ErrorPreventedResolution to ConstraintsNotSatisfiable?

Reply:
So I went and rechecked, and just above this if else block, the `Reason` and `Message` is being set regardless of whether the status is already in the expected state. So the answer to Joe's original questions is, no, it will set "Goodnight" and "Moon". 

However, I just realized that that if else block is a hangover from when `updateSubscriptionStatuses` was not separated out, so right now when we call that we're reaching out to the api server for all subscriptions regardless. So the `continue` here is unnecessary and looks like it's just creating more confusion. Adding a commit to remove that.

Reply:
Done.